frontend/src/components/content-component/product-card.tsx

```diff
@@ -1,42 +1,99 @@
-import React from 'react';
-import { Card, CardContent, Typography, CardMedia } from '@mui/material';
+import React, { useState } from 'react'
+import { Card, CardContent, Typography, CardMedia, Button } from '@mui/material'
 import './product-card.scss'
+import { styled } from '@mui/system'
 
 interface ProductCardProps {
-  id: number;
-  name: string;
-  price: number;
-  description: string;
-  image: string;
+  id: number
+  title: string
+  description: string | null
+  price_sgd: string | null
+  number_sold: number
+  category: string
+  link: string
+  image_url: string
 }
 
-const ProductCard: React.FC<ProductCardProps> = ({ id, image, name, description, price }) => {
+const CustomCard = styled(Card)``;
+
+const ProductCard: React.FC<ProductCardProps> = ({
+  id,
+  title,
+  description,
+  price_sgd,
+  number_sold,
+  category,
+  link,
+  image_url,
+}) => {
+
+  // useState and Functions to handle description displays
+  const TITLE_LENGTH_LIMIT = 75;
+  const DESCRIPTION_LENGTH_LIMIT = 200;
+  const [showFullText, setShowFullText] = useState(false);
+
+  const toggleText = () => {
+    setShowFullText(!showFullText);
+  };
+
+  const truncateText = (text: string | null, limit: number) => {
+    if (!text) return '';
+    return text.length > limit ? `${text.substring(0, limit)}...` : text;
+  };
+
+  const formatPrice = (price: string | null) => {
+    if (!price) return "Contact Seller for Price";
+    const prices = price.split(' - ');
+    return prices.length === 2
+      ? `$${prices[0]} - $${prices[1]}`
+      : `$${prices[0]}`;
+  };
+
+  const handleRedirect = () => {
+    window.open(link)
+  }
```

Comment:
You may want to:
1. Wrap constant definitions in `useMemo` hooks.
2. Wrap function definitions in `useCallback` hooks.

These reduce the number of definition refresh upon re-render and hence better performance. Constant/function declarations without hooks by default are refreshed on every re-render.

Reply:
@freddychenyouren2